apax/md/constraints.py

```diff
@@ -0,0 +1,33 @@
+from typing import Literal, Union
+
+import jax.numpy as jnp
+from pydantic import BaseModel, TypeAdapter
+
+
+class ConstraintBase(BaseModel):
+    def create(self, state):
+        pass
```

Comment:
here you use state, in FixAtoms you use `ref_state`.
Maybe use `abc`?
As this is a function factory, maybe add `-> Callable`

Reply:
Good point